### PR TITLE
[LibMan] Do not support multiple entries of the same library (of any version)

### PR DESCRIPTION
**Pull request by raquelsa**
This PR is against another PR branch dev/raquela/616997

### CODE REVIEW

**Comment by prafullbhosale** [comment]
This PR seems to contain changes that are already merged to `rel/15.8Preview4`
Were you trying to just get the changes from the `rel/15.8Preview4` branch into yours?

File: src/LibraryManager/LibrariesValidator.cs [hunk 1]
```diff
@@ -35,26 +35,34 @@ internal static class LibrariesValidator
...
             IEnumerable<ILibraryOperationResult> validateLibraries = await ValidatePropertiesAsync(libraries, dependencies, cancellationToken).ConfigureAwait(false);
-
             if (!validateLibraries.All(t => t.Success))
             {
                 return validateLibraries;
             }
 
-            IEnumerable<ILibraryOperationResult> expandLibraries= await ExpandLibrariesAsync(libraries, dependencies, defaultDestination, defaultProvider, cancellationToken).ConfigureAwait(false);
+            // Check for duplicate libraries 
+            if (GetDuplicateLibraries(libraries, dependencies, defaultProvider, cancellationToken) is List<string> duplicates && duplicates.Count > 0)
```
Comment (jimmylewis) [inline src/LibraryManager/LibrariesValidator.cs]: > is List<string> duplicates && duplicates.Count > 0 [](start = 98, length = 51)

.Any()

Comment (jimmylewis) [reply src/LibraryManager/LibrariesValidator.cs]: `duplicates` should probably be declared on the line before the if, something like:

````
IEnumerable<string> duplicates = GetDuplicateLibraries(...);
if(duplicates.Any())
{
    return new (...);
}
````

---
In reply to: [198657528](https://github.com/aspnet/LibraryManager/pull/220#discussion_r198657528) [](ancestors = 198657528)

File: src/LibraryManager/LibrariesValidator.cs [hunk 2]
```diff
@@ -35,26 +35,34 @@ internal static class LibrariesValidator
...
 
             libraries = expandLibraries.Select(l => l.InstallationState);
-            IEnumerable<FileConflict> fileConflicts = GetFilesConflicts(libraries, cancellationToken);
+            IEnumerable<FileConflict> fileConflicts = GetFilesConflicts(libraries);
             ILibraryOperationResult conflictErrors = GetConflictErrors(fileConflicts);
 
             return new [] { conflictErrors };
         }
 
+
```
Comment (jimmylewis) [inline src/LibraryManager/LibrariesValidator.cs]: nit: extra line

File: src/LibraryManager/LibrariesValidator.cs [hunk 3]
```diff
@@ -35,26 +35,34 @@ internal static class LibrariesValidator
...
             {
                 return validateLibraries;
             }
 
-            IEnumerable<ILibraryOperationResult> expandLibraries= await ExpandLibrariesAsync(libraries, dependencies, defaultDestination, defaultProvider, cancellationToken).ConfigureAwait(false);
+            // Check for duplicate libraries 
+            if (GetDuplicateLibraries(libraries, dependencies, defaultProvider, cancellationToken) is List<string> duplicates && duplicates.Count > 0)
+            {
+                return new[] { LibraryOperationResult.FromError(PredefinedErrors.DuplicateLibrariesInManifest(duplicates)) };
+            }
```
Comment (jimmylewis) [inline src/LibraryManager/LibrariesValidator.cs]: let's add one error for each duplicated library.

File: src/LibraryManager/LibrariesValidator.cs [hunk 4]
```diff
@@ -122,6 +130,66 @@ private static bool IsValidManifestVersion(string version)
             return validationStatus;
         }
 
+        private static IEnumerable<string> GetDuplicateLibraries(
```
Comment (jimmylewis) [inline src/LibraryManager/LibrariesValidator.cs]: >IEnumerable [](start = 23, length = 11)

return ISet<string>

File: test/libman.Test/LibmanInstallTest.cs [hunk 5]
```diff
@@ -169,10 +169,7 @@ public void TestInstall_Duplicate()
             int result = command.Execute("jquery@3.2.1", "--provider", "cdnjs");
 
             var testLogger = HostEnvironment.Logger as TestLogger;
-            Assert.AreEqual("Please specify a different destination.", testLogger.Messages.Last().Value);
-
```
Comment (jimmylewis) [inline test/libman.Test/LibmanInstallTest.cs]: Can you clean up in the CLI where this is shown?

**Pull request merged**